config/config.go

```diff
@@ -191,6 +196,19 @@ func (cfg Config) CallbackURL(path string, mergeParams ...url.Values) string {
 	return base.String()
 }
 
+// ValidWebhookURL returns true if the URL is an allowed webhook source.
+func (cfg Config) ValidWebhookURL(whURL string) bool {
+	if len(cfg.Webhook.AllowedURLs) == 0 {
+		return true
+	}
+	for _, u := range cfg.Webhook.AllowedURLs {
+		if strings.HasPrefix(whURL, u) {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
Webhooks commonly need auth info, this will break if just using prefix matching. Let's use a helper method (so we can help the referrer check too)

- account for implicit ports (e.g., `http` and `:80` as well as `https` and `:443`)
- ignore auth info, and reject it from config validation
- host is case-insensitive
- path is prefix checked itself
- query parameters are matched via the source URL

```go
func MatchURL(baseURL, testURL string) (bool, error) {

}
```

```go
func TestMatchURL(t *testing.T) {
	// true
	MatchURL("http://example.com/", "HTTP://example.COM")
	MatchURL("http://example.com/", "HTTP://foo:bar@example.COM:80")
	MatchURL("http://example.com/?notAllowedQueryParam=&requiredQueryParam=1", "http://example.com?requiredQueryParam=1")

	// false
	MatchURL("http://example.com/?notAllowedQueryParam=&requiredQueryParam=1", "http://example.com")
	MatchURL("http://example.com/?notAllowedQueryParam=", "http://example.com?notAllowedQueryParam=1")
	MatchURL("https://example.com", "http://example.com")
```

We can't really restrict paths more than we have since we already have the prefix precedent in referrer URLs (e.g., trailing `/`) so those can be just a direct match.

Handling the implicit ports & capitalization should also help with some auth referrer issues we've seen in the past.